timecache/first_seen_cache.go

```diff
@@ -1,72 +1,56 @@
 package timecache
 
 import (
-	"container/list"
+	"context"
 	"sync"
 	"time"
 )
 
-// FirstSeenCache is a thread-safe copy of https://github.com/whyrusleeping/timecache.
+// FirstSeenCache is a time cache that only marks the expiry of a message when first added.
 type FirstSeenCache struct {
-	q     *list.List
-	m     map[string]time.Time
-	span  time.Duration
-	guard *sync.RWMutex
-}
+	lk  sync.RWMutex
+	m   map[string]time.Time
+	ttl time.Duration
 
-func newFirstSeenCache(span time.Duration) TimeCache {
-	return &FirstSeenCache{
-		q:     list.New(),
-		m:     make(map[string]time.Time),
-		span:  span,
-		guard: new(sync.RWMutex),
-	}
+	done func()
 }
 
-func (tc FirstSeenCache) Add(s string) {
-	tc.guard.Lock()
-	defer tc.guard.Unlock()
+var _ TimeCache = (*FirstSeenCache)(nil)
 
-	_, ok := tc.m[s]
-	if ok {
-		log.Debug("first-seen: got same entry")
```

Comment:
Since we removed the log here, we can also remove the global log

Reply:
i guess... doesnt hurt all that much though.